system/lib/libc/dynlink.c

```diff
@@ -188,7 +231,7 @@ static void dlopen_js_onsuccess(struct dso* dso, struct async_data* data) {
       dso->mem_addr,
       dso->mem_size);
   load_library_done(dso);
-  do_write_lock();
+  do_write_unlock();
```

Comment:
This line made me sit back in my chair...

Is it a bugfix? Or how does it relate to this PR?

Reply:
Yeah, this just a typo from a previous change.   It seems we don't have any test coverage for async dlopen + threads yet.  Adding those can happen as a separate change I think.